java/ql/automodel/src/AutomodelApplicationModeExtractNegativeExamples.ql

```diff
@@ -48,18 +48,20 @@ from
 where
   endpoint = getSampleForCharacteristic(characteristic, 100) and
   extensibleType = endpoint.getExtensibleType() and
+  // the node is know not to be an endpoint of any appropriate type
+  forall(EndpointType tp | tp = endpoint.getAPotentialType() |
+    characteristic.hasImplications(tp, false, _)
```

Comment:
Your reasoning is correct, yes. As written, this maintains the previous behaviour of trying to find a single characteristic that rules this candidate out, but as you say it may well be that it is ruled out by a combination of several characteristics, in which case we'd have to change the logic a bit.